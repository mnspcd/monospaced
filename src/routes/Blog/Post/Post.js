import DocumentMeta from "react-document-meta";
import PropTypes from "prop-types";
import React from "react";

import Grid from "../../../components/Grid";
import Logotype from "../../../components/Logotype";
import Markdown, { Heading } from "../../../components/Markdown";
import Masthead from "../../../components/Masthead";
import Space from "../../../components/Space";

const Post = ({
  content: { blog, description, mastheadLinks, title },
  route: { path },
}) => {
  const post = blog.find(blogPost => blogPost.slug === path);
  const {
    mdx: {
      default: Mdx,
      meta: { date: postDate, title: postTitle },
    },
  } = post;

  return (
    <React.Fragment>
      <DocumentMeta
        title={`${postTitle} | ${title} Blog`}
        description={description}
      />
      <Space size="l">
        <Masthead
          activePath="/blog/"
          links={mastheadLinks}
          logo={<Logotype height="1.5em" />}
        />
      </Space>
      <Space>
        <Grid>
          <Grid.Item align="center" colSpan="1">
            <Heading size="m">{postDate}</Heading>
          </Grid.Item>
          <Grid.Item colSpan="4">
            <Heading level="1" size="xxxl">
              {postTitle}
            </Heading>
          </Grid.Item>
        </Grid>
      </Space>
      <Space size="xl">
        <Grid>
          <Grid.Item colSpan="4" colStart="2">
            <Markdown>
              <Mdx />
            </Markdown>
          </Grid.Item>
        </Grid>
      </Space>
    </React.Fragment>
  );
};

Post.propTypes = { content: PropTypes.object, route: PropTypes.object };

export default Post;
