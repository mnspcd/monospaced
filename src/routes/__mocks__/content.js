import React from "react";

/* eslint-disable react/display-name */
module.exports = {
  blog: [
    {
      mdx: {
        default: () => <div />,
        meta: {
          date: "2018-10-07",
          layout: () => <div />,
          title: "title",
        },
      },
      slug: "slug",
    },
  ],
  clients: [
    {
      href: "http://bladudflies.com",
      icon: "BladudFlies",
      label: "Bladud Flies!",
    },
    {
      href: "https://www.compass-group.co.uk",
      icon: "Chartwells",
      label: "Chartwells",
    },
    {
      href: "https://www.digitas.com",
      icon: "Digitas",
      label: "Digitas",
    },
    {
      href: "https://www.dowjones.com",
      icon: "DowJones",
      label: "Dow Jones",
    },
    {
      href: "https://www.findmypast.co.uk",
      icon: "FindMyPast",
      label: "Find My Past",
    },
    {
      href: "https://frfrm.com",
      icon: "Freeform",
      label: "Freeform Labs",
    },
    {
      href: "http://www.thegrandunion.com",
      icon: "GrandUnion",
      label: "Grand Union",
    },
    {
      href: "https://hanno.co",
      icon: "Hanno",
      label: "Hanno",
    },
    {
      href: "https://www.majorplayers.co.uk",
      icon: "MajorPlayers",
      label: "Major Players",
    },
    {
      href: "https://www.national-lottery.co.uk",
      icon: "NationalLottery",
      label: "The National Lottery",
    },
    {
      href: "https://www.nhs.uk",
      icon: "Nhs",
      label: "NHS",
    },
    {
      href: "https://www.ogilvy.com",
      icon: "Ogilvy",
      label: "Ogilvy",
    },
    {
      href: "http://www.social360monitoring.com",
      icon: "Social360",
      label: "Social 360",
    },
    {
      href: "https://www.stordis.com",
      icon: "Stordis",
      label: "Stordis",
    },
    {
      href: "http://wearesuburb.com",
      icon: "Suburb",
      label: "Suburb",
    },
    {
      href: "https://www.thomsonreuters.com",
      icon: "ThomsonReuters",
      label: "Thomson Reuters",
    },
  ],
  description: "Modern front-end web development",
  footerLinks: [
    {
      href: "/legal/",
      routerLink: true,
      text: "legal",
    },
    {
      href: "/lighthouse.report.html",
      text: "lighthouse",
    },
    {
      href: "/login/",
      text: "login",
    },
    {
      href: "http://localhost:9001",
      text: "storybook",
    },
    {
      href: "https://github.com/monospaced/monospaced.com",
      text: "view-source",
    },
  ],
  mastheadLinks: [
    {
      href: "/blog/",
      routerLink: true,
      text: "blog",
    },
    {
      href: "http://localhost:9001",
      text: "design-system",
    },
    {
      href: "mailto:info@monospaced.com",
      text: "info@monospaced.com",
    },
  ],
  title: "Monospaced",
};
/* eslint-enable */
