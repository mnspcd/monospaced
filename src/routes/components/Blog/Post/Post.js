import PropTypes from "prop-types";
import React from "react";
import DocumentMeta from "react-document-meta";

import {
  Grid,
  Heading,
  Logotype,
  Markdown,
  Masthead,
  Space,
} from "../../../../@monospaced/modern";

const Post = ({
  content: {
    blog: { posts },
    masthead: { links },
    meta: { description, title },
  },
  route: { path },
}) => {
  const {
    default: PostContent,
    meta: { date, title: postTitle },
  } = posts.find(post => post.meta.slug === path);

  return (
    <>
      <DocumentMeta
        title={`${postTitle} | ${title} Blog`}
        description={description}
      />
      <Masthead
        activePath="/blog/"
        links={links}
        logo={<Logotype height="1.5em" />}
      />
      <Space size="l" />
      <Grid>
        <Grid.Item align="center" colSpan="2">
          <Heading size="m">{date}</Heading>
        </Grid.Item>
        <Grid.Item colSpan="8">
          <Heading level="1" size="xxxl">
            {postTitle}
          </Heading>
        </Grid.Item>
      </Grid>
      <Space />
      <Grid>
        <Grid.Item colSpan="8" colStart="3">
          <Markdown>
            <PostContent />
          </Markdown>
        </Grid.Item>
      </Grid>
      <Space size="xl" />
    </>
  );
};

Post.propTypes = { content: PropTypes.object, route: PropTypes.object };

export default Post;
