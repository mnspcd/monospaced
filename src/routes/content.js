module.exports = {
  title: "Monospaced",
  description: "Modern web software solutions",
  blog: [], // require("../blog").default,
  clients: [
    {
      href: "http://bladudflies.com",
      icon: "BladudFlies",
    },
    {
      href: "https://www.compass-group.co.uk",
      icon: "Chartwells",
    },
    {
      href: "https://www.digitas.com",
      icon: "Digitas",
    },
    {
      href: "https://www.dowjones.com",
      icon: "DowJones",
    },
    {
      href: "https://www.findmypast.co.uk",
      icon: "FindMyPast",
    },
    {
      href: "https://frfrm.com",
      icon: "Freeform",
    },
    {
      href: "http://www.thegrandunion.com",
      icon: "GrandUnion",
    },
    {
      href: "https://hanno.co",
      icon: "Hanno",
    },
    {
      href: "https://www.majorplayers.co.uk",
      icon: "MajorPlayers",
    },
    {
      href: "https://www.national-lottery.co.uk",
      icon: "NationalLottery",
    },
    {
      href: "https://www.nhs.uk",
      icon: "Nhs",
    },
    {
      href: "https://www.ogilvy.com",
      icon: "Ogilvy",
    },
    {
      href: "https://www.social360monitoring.com",
      icon: "Social360",
    },
    {
      href: "https://www.stordis.com",
      icon: "Stordis",
    },
    {
      href: "http://wearesuburb.com",
      icon: "Suburb",
    },
    {
      href: "https://thomsonreuters.com",
      icon: "ThomsonReuters",
    },
  ],
  footerLinks: [
    {
      href: "/legal/",
      routerLink: true,
      text: "legal",
    },
    {
      href: "/lighthouse.report.html",
      text: "lighthouse",
    },
    {
      href: "/login/",
      text: "login",
    },
    {
      href: "http://localhost/storybook/",
      text: "storybook",
    },
    {
      href: "https://github.com/monospaced/monospaced.com",
      text: "view-source",
    },
  ],
  mastheadLinks: [
    /* {
      href: "/blog/",
      routerLink: true,
      text: "blog",
    },
    {
      href: "/design-system/",
      routerLink: true,
      text: "design-system",
    }, */
    {
      href: "mailto:info@monospaced.com",
      text: "info@monospaced.com",
    },
  ],
};
