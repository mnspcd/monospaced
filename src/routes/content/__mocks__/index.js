import React from "react";

/* eslint-disable react/display-name */
module.exports = {
  blog: {
    heading: "Blog",
    posts: [
      {
        default: () => <div />,
        meta: {
          date: "2018-10-07",
          layout: () => <div />,
          slug: "slug",
          title: "title",
        },
      },
    ],
  },
  clients: {
    heading: "Selected clients",
    list: [
      {
        href: "http://bladudflies.com",
        label: "Bladud Flies!",
        Logo: () => <div />,
      },
      {
        href: "https://www.compass-group.co.uk",
        label: "Chartwells",
        Logo: () => <div />,
      },
      {
        href: "https://www.digitas.com",
        label: "Digitas",
        Logo: () => <div />,
      },
      {
        href: "https://www.dowjones.com",
        label: "Dow Jones",
        Logo: () => <div />,
      },
      {
        href: "https://www.findmypast.co.uk",
        label: "Find My Past",
        Logo: () => <div />,
      },
      {
        href: "https://frfrm.com",
        label: "Freeform Labs",
        Logo: () => <div />,
      },
      {
        href: "http://www.thegrandunion.com",
        label: "Grand Union",
        Logo: () => <div />,
      },
      {
        href: "https://hanno.co",
        label: "Hanno",
        Logo: () => <div />,
      },
      {
        href: "https://www.majorplayers.co.uk",
        label: "Major Players",
        Logo: () => <div />,
      },
      {
        href: "https://www.national-lottery.co.uk",
        label: "The National Lottery",
        Logo: () => <div />,
      },
      {
        href: "https://www.nhs.uk",
        label: "NHS",
        Logo: () => <div />,
      },
      {
        href: "https://www.ogilvy.com",
        label: "Ogilvy",
        Logo: () => <div />,
      },
      {
        href: "http://www.social360monitoring.com",
        label: "Social 360",
        Logo: () => <div />,
      },
      {
        href: "https://www.stordis.com",
        label: "Stordis",
        Logo: () => <div />,
      },
      {
        href: "http://wearesuburb.com",
        label: "Suburb",
        Logo: () => <div />,
      },
      {
        href: "https://www.thomsonreuters.com",
        label: "Thomson Reuters",
        Logo: () => <div />,
      },
    ],
  },
  footer: {
    links: [
      {
        href: "/storybook/",
        text: "design-system",
      },
      {
        href: "/legal/",
        routerLink: true,
        text: "legal",
      },
      {
        href: "/lighthouse.report.html",
        text: "lighthouse",
      },
      {
        href: "/login/",
        text: "login",
      },
      {
        href: "https://github.com/monospaced/monospaced.com",
        text: "view-source",
      },
    ],
  },
  legal: {
    company: {
      content: require("./legal/company.mdx").default,
      heading: "Company details",
    },
    heading: "Legal matters",
    privacy: {
      content: require("./legal/privacy.mdx").default,
      heading: "Privacy policy",
    },
  },
  masthead: {
    links: [
      {
        href: "/blog/",
        routerLink: true,
        text: "blog",
      },
      {
        href: "/storybook/",
        text: "design-system",
      },
      {
        href: "mailto:hello@monospaced.com",
        text: "hello@monospaced.com",
      },
    ],
  },
  meta: {
    description: "Modern web software solutions",
    title: "Monospaced",
  },
  services: {
    heading: "Our services",
    list: [
      "Design systems engineering",
      "Front-end development",
      "Legacy code rescue",
      "Prototyping, concepts & MVPs",
      "Web applications",
    ],
  },
};
/* eslint-enable */
