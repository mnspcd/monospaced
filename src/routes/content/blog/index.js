export default [require("./2018-10-07-welcome.mdx")];
