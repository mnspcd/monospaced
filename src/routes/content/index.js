import {
  BladudFlies,
  Chartwells,
  Digitas,
  DowJones,
  FindMyPast,
  Freeform,
  GrandUnion,
  Hanno,
  MajorPlayers,
  NationalLottery,
  Nhs,
  Ogilvy,
  Social360,
  Stordis,
  Suburb,
  ThomsonReuters,
} from "./svg";

export default {
  blog: {
    heading: "Blog",
    posts: [], // require("./blog").default,
  },
  clients: {
    heading: "Selected clients",
    list: [
      {
        href: "http://bladudflies.com",
        label: "Bladud Flies!",
        Logo: BladudFlies,
      },
      {
        href: "https://www.compass-group.co.uk",
        label: "Chartwells",
        Logo: Chartwells,
      },
      {
        href: "https://www.digitas.com",
        label: "Digitas",
        Logo: Digitas,
      },
      {
        href: "https://www.dowjones.com",
        label: "Dow Jones",
        Logo: DowJones,
      },
      {
        href: "https://www.findmypast.co.uk",
        label: "Find My Past",
        Logo: FindMyPast,
      },
      {
        href: "https://frfrm.com",
        label: "Freeform Labs",
        Logo: Freeform,
      },
      {
        href: "http://www.thegrandunion.com",
        label: "Grand Union",
        Logo: GrandUnion,
      },
      {
        href: "https://hanno.co",
        label: "Hanno",
        Logo: Hanno,
      },
      {
        href: "https://www.majorplayers.co.uk",
        label: "Major Players",
        Logo: MajorPlayers,
      },
      {
        href: "https://www.national-lottery.co.uk",
        label: "The National Lottery",
        Logo: NationalLottery,
      },
      {
        href: "https://www.nhs.uk",
        label: "NHS",
        Logo: Nhs,
      },
      {
        href: "https://www.ogilvy.com",
        label: "Ogilvy",
        Logo: Ogilvy,
      },
      {
        href: "http://www.social360monitoring.com",
        label: "Social 360",
        Logo: Social360,
      },
      {
        href: "https://www.stordis.com",
        label: "Stordis",
        Logo: Stordis,
      },
      {
        href: "http://wearesuburb.com",
        label: "Suburb",
        Logo: Suburb,
      },
      {
        href: "https://www.thomsonreuters.com",
        label: "Thomson Reuters",
        Logo: ThomsonReuters,
      },
    ],
  },
  footer: {
    links: [
      {
        href: "/legal/",
        routerLink: true,
        text: "legal",
      },
      // {
      //   href: "/lighthouse.report.html",
      //   text: "lighthouse",
      // },
      {
        href: "/login/",
        text: "login",
      },
      {
        href: "https://github.com/mnspcd/monospaced.com",
        text: "view-source",
      },
    ],
  },
  legal: {
    company: {
      content: require("./legal/company.mdx").default,
      heading: "Company details",
    },
    heading: "Legal matters",
    privacy: {
      content: require("./legal/privacy.mdx").default,
      heading: "Privacy policy",
    },
  },
  masthead: {
    links: [
      /* {
        href: "/blog/",
        routerLink: true,
        text: "blog",
      }, */
      {
        href: "/design-system/?path=/docs/introduction--page",
        text: "design-system",
      },
      {
        href: "mailto:hello@monospaced.com",
        text: "hello@monospaced.com",
      },
    ],
  },
  meta: {
    description: "Modern web software solutions",
    title: "Monospaced",
  },
  services: {
    heading: "Our services",
    list: [
      "Design systems engineering",
      "Front-end development",
      "Legacy code rescue",
      "Prototyping, concepts & MVPs",
      "Web applications",
    ],
  },
};
