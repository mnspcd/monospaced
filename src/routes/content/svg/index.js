import BladudFlies from "./bladud-flies.svg";
import Chartwells from "./chartwells.svg";
import Digitas from "./digitas.svg";
import DowJones from "./dow-jones.svg";
import FindMyPast from "./find-my-past.svg";
import Freeform from "./freeform.svg";
import GrandUnion from "./grand-union.svg";
import Hanno from "./hanno.svg";
import Hubble from "./hubble.svg";
import MajorPlayers from "./major-players.svg";
import NationalLottery from "./national-lottery.svg";
import Nhs from "./nhs.svg";
import Ogilvy from "./ogilvy.svg";
import Social360 from "./social-360.svg";
import Stordis from "./stordis.svg";
import ThomsonReuters from "./thomson-reuters.svg";

export {
  BladudFlies,
  Chartwells,
  Digitas,
  DowJones,
  FindMyPast,
  Freeform,
  GrandUnion,
  Hanno,
  Hubble,
  MajorPlayers,
  NationalLottery,
  Nhs,
  Ogilvy,
  Social360,
  Stordis,
  ThomsonReuters,
};
