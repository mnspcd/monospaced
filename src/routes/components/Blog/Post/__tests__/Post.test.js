/* eslint-env jest */
import { shallow } from "enzyme";
import React from "react";

import content from "../../../../content/__mocks__";

import Post from "../";

describe("Post component", () => {
  it("should render correctly", () => {
    const component = shallow(
      <Post content={content} route={{ path: "slug" }} />,
    );
    expect(component).toMatchSnapshot();
  });
});
