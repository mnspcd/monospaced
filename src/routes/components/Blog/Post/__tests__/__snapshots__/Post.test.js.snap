// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post component should render correctly 1`] = `
<Fragment>
  <SideEffect(DocumentMeta)
    description="Modern web software solutions"
    title="title | Monospaced Blog"
  />
  <Masthead
    activePath="/blog/"
    links={
      Array [
        Object {
          "href": "/blog/",
          "routerLink": true,
          "text": "blog",
        },
        Object {
          "href": "/storybook/",
          "text": "design-system",
        },
        Object {
          "href": "mailto:hello@monospaced.com",
          "text": "hello@monospaced.com",
        },
      ]
    }
    logo={
      <Logotype
        height="1.5em"
      />
    }
  />
  <Space
    size="l"
  />
  <Grid>
    <Grid.Item
      align="center"
      colSpan="2"
    >
      <Heading
        size="m"
      >
        2018-10-07
      </Heading>
    </Grid.Item>
    <Grid.Item
      colSpan="8"
    >
      <Heading
        level="1"
        size="xxxl"
      >
        title
      </Heading>
    </Grid.Item>
  </Grid>
  <Space />
  <Grid>
    <Grid.Item
      colSpan="8"
      colStart="3"
    >
      <Markdown>
        <_default />
      </Markdown>
    </Grid.Item>
  </Grid>
  <Space
    size="xl"
  />
</Fragment>
`;
